openssl/src/pkey_ctx.rs

```diff
@@ -920,6 +958,20 @@ mod test {
         ctx.keygen().unwrap();
     }
 
+    #[test]
+    #[cfg(not(boringssl))]
+    fn dsa_keygen() {
+        let mut ctx = PkeyCtx::new_id(Id::DSA).unwrap();
+        ctx.paramgen_init().unwrap();
+        ctx.set_dsa_paramgen_bits(2048).unwrap();
+        let params = ctx.paramgen().unwrap();
+        assert_eq!(params.size(), 64);
+
+        let mut key_ctx = PkeyCtx::new(&params).unwrap();
+        key_ctx.keygen_init().unwrap();
+        key_ctx.keygen().unwrap();
```

Comment:
No need for this

Reply:
No need to generate the key at all, the size assert is enough

Reply:
Removed